ocaml/xapi/xapi.ml

```diff
@@ -444,7 +447,60 @@ let attempt_pool_hello my_ip =
               [localhost_uuid] ;
             Some Permanent
         | `ok ->
-            None
+            let obj_uuid = Helpers.get_localhost_uuid () in
+            let message_name, _ =
+              Api_messages
+              .xapi_startup_blocked_as_version_higher_than_coordinator
+            in
+            let existing_alerts =
+              Client.Client.Message.get_all_records ~rpc ~session_id
+              |> List.filter (fun (_ref, record) ->
+                     record.API.message_obj_uuid = obj_uuid
+                     && record.API.message_name = message_name
+                 )
+            in
+            let compare_xapi_version_with_coordinator ~__context =
+              let coordinator_ref = Helpers.get_master ~__context in
+              let coordinator_xapi_ver =
+                Client.Client.Host.get_software_version ~rpc ~session_id
```

Comment:
At this point, we are able to use the `Db` module to more efficiently read from the DB. This is more efficient than doing a `Client` call, which makes a new connection.

Reply:
In fact, `Helpers.get_master` also uses the `Db` module internally, and you're doing it below as well.